pandas/io/excel/_openpyxl.py

```diff
@@ -531,18 +530,12 @@ def _convert_cell(self, cell, convert_float: bool) -> Scalar:
         return cell.value
 
     def get_sheet_data(self, sheet, convert_float: bool) -> List[List[Scalar]]:
-        # GH 39001
-        # Reading of excel file depends on dimension data being correct but
-        # writers sometimes omit or get it wrong
```

Comment:
Can you move this comment down to the `if is_readonly` below, add that it only impacts read-only worksheets